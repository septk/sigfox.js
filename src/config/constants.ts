export const API_CONFIG = {
  baseURL: 'https://api.sigfox.com/v2',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json-patch+json'
  }
}
