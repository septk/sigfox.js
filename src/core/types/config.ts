export interface IConfiguration {
  baseURL: 'https://api.sigfox.com/v2'
  headers: IHeaders
}

export interface IHeaders {
  Accept: string
  'Content-Type': string
}
