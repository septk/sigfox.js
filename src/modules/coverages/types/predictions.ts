export interface GlobalPredictions {
  locationCovered: boolean
  margins: [number]
}

export interface Redundancy {
  redundancy: number
}
