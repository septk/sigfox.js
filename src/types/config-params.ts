export interface ConfigParams {
  username: string
  password: string
}
