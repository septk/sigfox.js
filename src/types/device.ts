export interface Device {
  id: string
  name: string
  lastCom: number
  state: number
  comState: number
  location: Location
  deviceType: DeviceType
  group: DeviceType
  lqi: number
  activationTyime: number
  token: TokenStruct
  contract: DeviceType
  creationTime: number
  modemCertificate: DeviceType
  prototype: boolean
  automaticRenewal: boolean
  automaticRenewalStatus: number
  createdBy: string
  lastEditionTime: number
  lastEditedBy: string
  activable: boolean
}

export interface Location {
  lat: number
  lng: number
}

export interface DeviceType {
  id: string
}

export interface TokenStruct {
  state: number
  detailMessage: string
  end: number
}
