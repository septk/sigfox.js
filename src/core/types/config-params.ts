import { IConfiguration } from './config'

export interface ConfigParams {
  username: string
  password: string
}

export interface CustomParams {
  auth: ConfigParams
  apiConfig: IConfiguration
}
