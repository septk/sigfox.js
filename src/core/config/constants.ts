import { IConfiguration } from '../types/config'

export const API_CONFIG: IConfiguration = {
  baseURL: 'https://api.sigfox.com/v2',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json-patch+json'
  }
}
