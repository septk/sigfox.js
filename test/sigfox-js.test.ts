import * as nock from 'nock'

import SigfoxApi from '../src/sigfox-js'
import { ConfigParams } from '../src/types/config-params'
import { API_CONFIG } from '../src/config/constants'

/**
 * SigfoxApi test
 */

describe('SigfoxApi test', () => {
  it('should be a class', () => {
    const sigfox = new SigfoxApi({} as ConfigParams)
    expect(sigfox instanceof SigfoxApi).toBeTruthy()
  })

  describe('constructor', () => {
    it('SigfoxApiWrapper is instantiable with a config', () => {
      expect(new SigfoxApi({} as ConfigParams)).toBeInstanceOf(SigfoxApi)
    })
  })

  describe('getDeviceInfo() call', () => {
    it('should return device obj', async () => {
      nock(API_CONFIG.baseURL)
        .get(`/devices/9E1F98`)
        .reply(200)
      const sigfox = new SigfoxApi({} as ConfigParams)
      const result = await sigfox.getDeviceInfo('9E1F98')
      expect(result).toMatchSnapshot()
    })

    it('should return error', async () => {
      const sigfox = new SigfoxApi({} as ConfigParams)
      try {
        nock(API_CONFIG.baseURL)
          .get(`/devices/9E1F98`)
          .reply(404)
        await sigfox.getDeviceInfo('9E1F98')
      } catch (error) {
        expect(error).toMatchSnapshot()
      }
    })
  })
})
