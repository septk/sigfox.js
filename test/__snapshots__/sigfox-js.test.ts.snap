// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigfoxApi test getDeviceInfo() call should return device obj 1`] = `""`;
